### PR TITLE
Added IPv6 support

### PR DESCRIPTION
**Pull request by triadcode**

### CODE REVIEW

**Comment by adamgilbert** [comment]
Should we ignore (with a shellcheck comment) or address those shellcheck errors?

**Comment by triadcode** [comment]
> 

I believe we should fix the shellcheck errors but it would also be beneficial to refactor a few things in that script. For instance, the function `get_local_ip_addresses` does the same thing in 3 different ways even though the last approach (the one on the else part of the if statement) is valid for every Linux distribution and should be enough. In addition, the `filter_sort_local_ip_addresses`  doesn't seem to be called elsewhere either. Consequently, one might argue that it would be better to have its code in the first function.
The `get_local_ip_addresses` function could be then rewritten as something like `awk '/32 host/ { print "\tinet " f } {f=$2}' </proc/net/fib_trie | awk -F '[ \t]+|/' '{print $3}' | grep -v "127" | sort -Vu`

**Review by adamgilbert — APPROVED** [review]

File: shtdlib.sh [hunk 1]
```diff
@@ -158,30 +154,28 @@ function url_encode {
...
-    local -a local_iv4
-    if whichs ip ; then
-        ip -4 addr show | filter_sort_local_ip_addresses
-    elif whichs ifconfig ; then
-        ifconfig | filter_sort_local_ip_addresses
-    else
-        awk '/32 host/ { print "inet " f } {f=$2}' </proc/net/fib_trie | \
-            filter_sort_local_ip_addresses
-    fi
+      # Print non-loopback IPv4 addresses
```
Comment (hrpatel) [inline shtdlib.sh:159]: Did you mean to indent so much?

Comment (triadcode) [reply shtdlib.sh:159]: Nope, let me run a linter on it.

**Pull request merged**